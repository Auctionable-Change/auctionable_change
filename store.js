import React, { createContext, useContext, useReducer } from "react";

const StoreContext = createContext();

const initialState = {
  currentListing: {
    id: null,
    title: "",
    description: "",
    price: null,
    donor: "",
    donor_email: "",
    status: "",
    category: "",
    charity: "",
    charity_url: "",
    charity_score: "",
    charity_score_image: "",
    image: "",
    bids: [],
  },
  buyerDetails: {
    item_id: null,
    bidder_name: "",
    bidder_email: "",
    amount: null,
    street_address: "",
    city: "",
    state: "",
    zip_code: "",
    receipt: "",
  },
  listingToPost: {},
};

const reducer = (state, action) => {
  switch (action.type) {
    case "ADD_CURRENT_LISTING":
      return {
        ...state,
        currentListing: action.currentListing,
      };
    case "ADD_SELECTED_CHARITY":
      return {
        ...state,
        listingToPost: {
          ...state.listingToPost,
          charity: action.selectedCharity.name,
          charity_url: action.selectedCharity.url,
          charity_score: action.selectedCharity.rating,
          charity_score_image: action.selectedCharity.rating_image,
        },
      };
    case "ADD_BUYER_DETAILS":
      return {
        ...state,
        buyerDetails: { ...state.buyerDetails, ...action.buyerDetails },
      };
    case "ADD_TO_LISTING":
      return {
        ...state,
        listingToPost: { ...state.listingToPost, ...action.listingToPost },
      };
    case "POST_BID":
      return {
        ...state,
        currentBid: action.bid
      };
      case "SET_USER":
      return {
        ...state,
        loginDetails: action.loginDetails,
        userInfo: action.userInfo,
        registrationDetails: action.registration
      }

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <StoreContext.Provider value={{ state, dispatch }}>
      {children}
    </StoreContext.Provider>
  );
};

export const useStore = () => useContext(StoreContext);
