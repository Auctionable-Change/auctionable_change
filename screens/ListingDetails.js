import * as React from "react";
import { Image, Text, View } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import NavBar from "../components/NavBar/NavBar";
import { StyleSheet } from "react-native";
import { ScrollView } from "react-native-gesture-handler";
import { useStore } from "../store";
import * as Linking from "expo-linking";
import { Button } from "native-base";

const ListingDetails = ({ navigation }) => {
  const { state } = useStore();
  return (
    <SafeAreaView style={styles.container}>
      <ScrollView contentContainerStyle={styles.scroll}>
        <View style={styles.itemContainer}>
          <Text style={styles.title}>{state.currentListing.title}</Text>
          <Image
            style={styles.image}
            source={
              state.currentListing.image
                ? { uri: state.currentListing.image }
                : require("../assets/icons/no-photo-selected.png")
            }
          />
          <Text style={styles.description}>
            {state.currentListing.description}
          </Text>
          <Text style={{ fontSize: 20 }}>
            Listed Price: ${state.currentListing.price}
          </Text>
          <Text style={styles.charityDescription}>
            Purchase of this item supports {state.currentListing.charity}
          </Text>
          <Text>Charity Score: {state.currentListing.charity_score}</Text>
          <View style={{ flexDirection: "row", flexWrap: "wrap" }}>
            <Text>Read more about their mission </Text>
            <Text
              style={{ fontWeight: "bold" }}
              onPress={() => Linking.openURL(state.currentListing.charity_url)}
            >
              here.
            </Text>
          </View>
        </View>
        <Button block success onPress={() => navigation.navigate("Purchase")}>
          <Text style={{ color: "white", fontSize: 15 }}>Donate For Item</Text>
        </Button>
        <View style={styles.charityContainer}>
          <Text style={styles.charityAttribution}>
            Charity information provided by
          </Text>
          <Image
            source={require("../assets/charity_navigator.jpg")}
            style={{ width: 90, height: 45 }}
          />
        </View>
      </ScrollView>
      <NavBar navigation={navigation} />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
  },
  itemContainer: {
    backgroundColor: "white",
    padding: 15,
    borderRadius: 10,
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
  },
  title: {
    margin: 10,
    fontSize: 30,
    textAlign: "center",
    fontWeight: "bold",
  },
  scroll: {
    display: "flex",
    alignItems: "center",
    padding: 15,
  },
  image: {
    width: 225,
    height: 225,
    resizeMode: "cover",
    alignSelf: "center",
  },
  description: {
    marginTop: 15,
    marginBottom: 15,
  },
  charityDescription: {
    marginTop: 15,
  },
  charityAttribution: {
    fontStyle: "italic",
    width: 140,
    textAlign: "center",
    color: "#336799",
    marginTop: 5,
    marginBottom: 30,
  },
  charityContainer: {
    flexWrap: "wrap",
    flexDirection: "row",
    marginTop: 15,
    height: 120,
  },
});

export default ListingDetails;
