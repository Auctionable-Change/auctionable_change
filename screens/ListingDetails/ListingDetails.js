import * as React from "react";
import { Image, Text, View } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import NavBar from "../../components/NavBar/NavBar";
import { StyleSheet } from "react-native";
import { ScrollView } from "react-native-gesture-handler";
import { useStore } from "../../store";
import * as Linking from "expo-linking";
import { Button } from "native-base";

const ListingDetails = ({ navigation }) => {
  const { state } = useStore();
  return (
    <SafeAreaView style={styles.container}>
      <ScrollView contentContainerStyle={styles.scroll}>
        <Text style={styles.title}>{state.currentListing.title}</Text>
        <Image
          style={styles.image}
          source={
            state.currentListing.image
              ? { uri: state.currentListing.image }
              : require("../../assets/icons/no-photo-selected.png")
          }
        />
        <Text style={styles.description}>
          {state.currentListing.description}
        </Text>
        <Text style={styles.description}>
          This item supports {state.currentListing.charity}
        </Text>
        <Text style={styles.description}>
          Minimum Donation: ${state.currentListing.price}
        </Text>
        <Text style={styles.description}>
          Charity Score: {state.currentListing.charity_score}
        </Text>
        <View style={{ flexDirection: "row", flexWrap: "wrap" }}>
          <Text style={{ fontSize: 15 }}>Read more about their mission </Text>
          <Text
            style={{ fontWeight: "bold", textDecorationLine: "underline" }}
            onPress={() => Linking.openURL(state.currentListing.charity_url)}
          >
            here
          </Text>
          <Text style={styles.description}>
            Charity Score: {state.currentListing.charity_score}
          </Text>
          <View style={{ flexDirection: "row", flexWrap: "wrap" }}>
            <Text style={{ fontSize: 15, marginLeft: 5, fontFamily: "quicksand" }}>
              Read more about their mission{" "}
            </Text>
            <Text
              style={{ textDecorationLine: "underline", fontFamily: "quicksand-bold" }}
              onPress={() => Linking.openURL(state.currentListing.charity_url)}
            >
              here
            </Text>
          </View>
        </View>
        <Button block success style={{ margin: 5 }} onPress={() => navigation.navigate("Purchase")}>
          <Text style={{ color: "white", fontSize: 15, fontFamily: "quicksand-bold" }}>Donate For Item</Text>
        </Button>
        <View style={styles.charityContainer}>
          <Text style={styles.charityAttribution}>
            Charity information provided by
          </Text>
          <Image
            source={require("../../assets/charity_navigator.jpg")}
            style={{ width: 90, height: 45 }}
          />
        </View>
      </ScrollView>
      <NavBar navigation={navigation} />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#FFFFFF",
  },
  title: {
    marginBottom: 5,
    fontSize: 25,
    textAlign: "center",
    fontFamily: "quicksand-bold"
  },
  scroll: {
    margin: 0,
    display: "flex",
    alignItems: "center",
  },
  image: {
    width: "100%",
    height: "100%",
    resizeMode: "cover",
    alignSelf: "center",
    margin: 5,
  },
  description: {
    margin: 5,
    fontSize: 15,
    fontFamily: "quicksand"
  },
  charityAttribution: {
    width: 140,
    textAlign: "center",
    color: "#336799",
    margin: 5,
    fontFamily: "quicksand"
  },
  charityContainer: {
    flexWrap: "wrap",
    flexDirection: "row",
    marginTop: 5,
    marginBottom: 15,
    height: 120,
  },
});

export default ListingDetails;
