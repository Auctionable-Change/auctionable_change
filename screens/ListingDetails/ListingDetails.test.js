import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import ListingDetails from './ListingDetails';
import { StoreProvider } from '../../store'
import {expect, it} from '@jest/globals';
import renderer from "react-test-renderer";

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');

describe('ListingDetails', () => {
  it('should render', () => {
    const tree = renderer.create(
      <StoreProvider>
        <ListingDetails/>
      </StoreProvider>).toJSON();
   expect(tree).toMatchSnapshot();
  })
})