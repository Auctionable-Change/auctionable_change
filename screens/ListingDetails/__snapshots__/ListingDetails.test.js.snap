// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingDetails should render 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    collapsable={false}
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 10,
            "margin": 15,
            "padding": 15,
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.18,
            "shadowRadius": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "bold",
              "margin": 10,
              "textAlign": "center",
            }
          }
        >
          
        </Text>
        <Image
          source={1}
          style={
            Object {
              "alignSelf": "center",
              "height": 225,
              "resizeMode": "cover",
              "width": 225,
            }
          }
        />
        <Text
          style={
            Object {
              "marginBottom": 15,
              "marginTop": 15,
            }
          }
        >
          
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
            }
          }
        >
          Listed Price: $
        </Text>
        <Text
          style={
            Object {
              "marginTop": 15,
            }
          }
        >
          Purchase of this item supports 
          
        </Text>
        <Text>
          Charity Score: 
        </Text>
        <View
          style={
            Object {
              "flexDirection": "row",
              "flexWrap": "wrap",
            }
          }
        >
          <Text>
            Read more about their mission 
          </Text>
          <Text
            onPress={[Function]}
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            here.
          </Text>
        </View>
      </View>
      <RNGestureHandlerButton
        collapsable={false}
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            undefined,
          ]
        }
      >
        <View
          accessible={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "marginTop": 5,
              "opacity": 1,
              "padding": 10,
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.18,
              "shadowRadius": 1,
              "width": 200,
            }
          }
        >
          <Text>
            Donate For Item
          </Text>
        </View>
      </RNGestureHandlerButton>
      <View
        style={
          Object {
            "flexDirection": "row",
            "flexWrap": "wrap",
            "height": 120,
            "marginTop": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#336799",
              "fontStyle": "italic",
              "marginBottom": 30,
              "marginTop": 5,
              "textAlign": "center",
              "width": 140,
            }
          }
        >
          Charity information provided by
        </Text>
        <Image
          source={1}
          style={
            Object {
              "height": 45,
              "width": 90,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "bottom": "0%",
        "display": "flex",
        "flexDirection": "row",
        "height": "11%",
        "justifyContent": "space-around",
        "padding": 10,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
      >
        <Image
          accessibilityRole="button"
          id="search"
          source={1}
          style={
            Object {
              "height": 45,
              "width": 45,
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
      >
        <Image
          accessibilityRole="button"
          id="add"
          source={1}
          style={
            Object {
              "height": 45,
              "width": 45,
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
      >
        <Image
          accessibilityRole="button"
          id="profile"
          source={1}
          style={
            Object {
              "height": 45,
              "width": 45,
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
  </View>
</RNCSafeAreaView>
`;
